jupyterlab/commands.py

```diff
@@ -1628,7 +1629,7 @@ def _node_check(logger):
         output = subprocess.check_output([node, 'node-version-check.js'], cwd=HERE)
         logger.debug(output.decode('utf-8'))
     except Exception:
-        data = _get_core_data()
+        data = CoreConfig().data
         ver = data['engines']['node']
```

Comment:
For example, we expect there to be 'engines': 'node' metadata.

Reply:
The content of `data` is supposed to be a lab internal detail. As an alternative, I could call the method `._data`, to underline the fact? Or `._get_data_internal()` ?

Reply:
> The content of data is supposed to be a lab internal detail

C.f. its docstring.

Reply:
Ah, so your custom spin doesn't use our AppHandler, right?

Reply:
It does, but it only modifies `data` via the methods on `CoreConfig`. As such, the internal detail fields like `engine` are left untouched.

Reply:
I'm on my phone, but doesn't that metadata still need to be there then?  As in, we need to document the minimum fields that need to be present?

Reply:
Note that the data is originally filled with the data from `staging/package.json`. This public API only allows for modifying the extensions/mimeExtensions/singletons. The `data` property (getter only) is meant to be used by whoever consumes the core config (internal only). Technically, since we are returning the inner data by reference, a user could modify the content of `data` in place. The behavior in this case is undefined (by spec), and can be considered similar in nature to monkey patching. We could try to make this harder to do by accident by doing a deepcopy on access, but I'm not sure if that is worth the hassle.

Reply:
It feels off to me that they'd have to use our package
json directly as a starting point.  I'd rather they had to provide their own metadata the met a certain spec.